src/main/scala/cromwell/engine/backend/local/LocalFileSystemOperations.scala

```diff
@@ -0,0 +1,169 @@
+import scala.util.{Failure, Success, Try}
+
+trait LocalFileSystemOperations {
+  def postProcess(backendCall: LocalFileSystemBackendCall): Try[CallOutputs] = {
+    // Evaluate output expressions, performing conversions from String -> File where required.
+    val outputMappings = backendCall.call.task.outputs.map { taskOutput =>
+      val tryConvertedValue =
+        for {
+          expressionValue <- taskOutput.expression.evaluate(backendCall.lookupFunction, backendCall.engineFunctions, interpolateStrings=true)
+          convertedValue <- outputAutoConversion(backendCall, taskOutput, expressionValue)
+          pathAdjustedValue <- Success(absolutizeOutputWdlFile(convertedValue, backendCall.callRootPath))
+        } yield pathAdjustedValue
+      taskOutput.name -> tryConvertedValue
+    }
+
+    val taskOutputFailures = outputMappings.filter { _._2.isFailure }
+
+    if (taskOutputFailures.isEmpty) {
+      val unwrappedMap = outputMappings.collect { case (name, Success(wdlValue)) => name -> wdlValue }.toMap
+      Success(unwrappedMap)
+    } else {
+      val message = taskOutputFailures.collect { case (name, Failure(e)) => s"$name: $e" }.mkString("\n")
+      Failure(new Throwable(s"Workflow ${backendCall.workflowDescriptor.id}: $message"))
+    }
+  }
+
+  def adjustOutputPaths(call: Call, outputs: CallOutputs): CallOutputs = outputs
+
+  def stdoutStderr(workflowId: WorkflowId, workflowName: String, callName: String): StdoutStderr = {
+    val dir = LocalBackend.hostCallPath(workflowName, workflowId, callName)
+    StdoutStderr(
+      stdout = WdlFile(dir.resolve("stdout").toAbsolutePath.toString),
+      stderr = WdlFile(dir.resolve("stderr").toAbsolutePath.toString)
+    )
+  }
+
+  /**
+   * Creates host execution directory, inputs path, and outputs path.  Stages any input files into the workflow-inputs
+   * directory and localizes their paths relative to the container.
+   */
+  def initializeForWorkflow(descriptor: WorkflowDescriptor): HostInputs = {
+    val hostExecutionDirectory = LocalBackend.hostExecutionPath(descriptor).toFile
+    hostExecutionDirectory.mkdirs()
+    val hostExecutionAbsolutePath = hostExecutionDirectory.getAbsolutePath
+    Array("workflow-inputs", "workflow-outputs") foreach { Paths.get(hostExecutionAbsolutePath, _).toFile.mkdir() }
+    stageWorkflowInputs(descriptor)
+  }
+
+  /**
+   * Return a possibly altered copy of inputs reflecting any localization of input file paths that might have
+   * been performed for this `Backend` implementation.
+   */
+  def adjustInputPaths(call: Call, inputs: CallInputs): CallInputs = {
+    // If this call is using Docker, adjust input paths, otherwise return unaltered input paths.
+    def adjustPath(nameAndValue: (String, WdlValue)): (String, WdlValue) = {
+      val (name, value) = nameAndValue
+      val adjusted = value match {
+        case WdlFile(path) =>
+          // Host path would look like cromwell-executions/three-step/f00ba4/call-ps/stdout.txt
```

Comment:
I think it's time for a function here. Please make a function like `adjustPathForWdlFile` and pull it out of the match